plugins/out_opensearch/opensearch.c

```diff
@@ -875,6 +998,18 @@ static void cb_opensearch_flush(struct flb_event_chunk *event_chunk,
         FLB_OUTPUT_RETURN(FLB_RETRY);
     }
 
+	/*
+	 * create index template if data stream mode is enabled
+	*/
+
+	if (ctx->data_stream_mode == FLB_TRUE) {
+		res = create_index_template(ctx);
```

Comment:
Putting on every request of flush is insufficient to handle on this.
We have to search the existing template if possible and not to overwrite by default.

Reply:
I'll look to update the logic, will introduce a `Template_Overwrite` flag in fluent-bit opensearch config which will be set to false by default. Now, if a template with the given name exists, overwrite only if the flag is set to true in the output config. Wdyt?

Reply:
Yup. Your understanding is correct. Thanks for your understanding.